minecraft/networking/connection.py

```diff
@@ -436,14 +436,23 @@ def read_packet(self, stream):
 
             if self.connection.options.compression_enabled:
                 decompressed_size = VarInt.read(packet_data)
-                if decompressed_size > 0:
-                    decompressed_packet = decompress(packet_data.read())
-                    assert len(decompressed_packet) == decompressed_size, \
-                        'decompressed length %d, but expected %d' % \
-                        (len(decompressed_packet), decompressed_size)
-                    packet_data.reset()
-                    packet_data.send(decompressed_packet)
-                    packet_data.reset_cursor()
+                rawPacketData = None
+                try:
```

Comment:
Unfortunately I don't know anything in specific about the plugins used nor the server version (afaik they use 1.8.8).
In general this faulty behaviour can be reproduced on play(.)saicopvp(.)com

I did some further investigation and the server claims to support the following protocol versions:
[47, 107, 108, 109, 110, 210, 301, 302, 303, 304, 305, 306, 307, 308, 309, 310, 311, 312, 313, 314, 315, 316, 317, 318, 319, 320, 321, 322, 323, 324, 325, 326, 327, 328, 329, 330, 331, 332, 333, 334, 335, 336, 337, 338]

However when connecting with ``self.context.protocol_version = max(self.allowed_proto_versions)`` the client doesn't do anything.
After some debugging I found out that the server sends the following: ``0x1A 0x1A DisconnectPacketPlayState {'json_data': '"Unsupported protocol version 335"'}`` or similar depending on the version the client connects with.

In general this is messed up from the start already. The server claims to support this specific version and when connecting it sends a DisconnectPacketPlayState-Packet.

Regardless for my testing I adjusted the way how the version is determined by using:
``self.context.protocol_version = min(self.allowed_proto_versions)`` 

this will result in:
``  File "~/pyCraft/minecraft/networking/connection.py", line 440, in read_packet
    decompressed_packet = decompress(packet_data.read())
zlib.error: Error -5 while decompressing data: incomplete or truncated stream
``
If I add output about the `decompressed_size` its a non-zero value and dynamic for each packet. this it will try to decompress the packet and fail with the previous error.

I can't reproduce this in any way with any paper spigot / spigot server with ViaVersion.